# HG changeset patch
# User Rik <rik@octave.org>
# Date 1504586056 25200
#      Mon Sep 04 21:34:16 2017 -0700
# Node ID 6bba9e4a1e8d66a1c6d0f7d0dadd359bb885367f
# Parent  94cbb6dd7b2114b22bc9a1ed5ce9469b411bc9b6
bicgstab.m, cgs.m: Fix typo preventing use of string input (bug #51219).

* bicgstab.m, cgs.m: If input is string, use str2func to create a function
handle and assign it to variable "Ax", not input "A".

diff --git a/scripts/sparse/bicgstab.m b/scripts/sparse/bicgstab.m
--- a/scripts/sparse/bicgstab.m
+++ b/scripts/sparse/bicgstab.m
@@ -72,17 +72,17 @@
 function [x, flag, relres, iter, resvec] = bicgstab (A, b, rtol, maxit,
                                                      M1, M2, x0)
 
   if (nargin < 2 || nargin > 7 || ! isvector (full (b)))
     print_usage ();
   endif
 
   if (ischar (A))
-    A = str2func (A);
+    Ax = str2func (A);
   elseif (isnumeric(A) && issquare (A))
     Ax = @(x) A  * x;
   elseif (isa (A, "function_handle"))
     Ax = @(x) feval (A, x);
   else
     error ("bicgstab: A must be a square matrix or function");
   endif
 
diff --git a/scripts/sparse/cgs.m b/scripts/sparse/cgs.m
--- a/scripts/sparse/cgs.m
+++ b/scripts/sparse/cgs.m
@@ -68,17 +68,17 @@
 ## @seealso{pcg, bicgstab, bicg, gmres, qmr}
 ## @end deftypefn
 
 function [x, flag, relres, iter, resvec] = cgs (A, b, rtol, maxit, M1, M2, x0)
 
   if (nargin >= 2 && nargin <= 7 && isvector (full (b)))
 
     if (ischar (A))
-      A = str2func (A);
+      Ax = str2func (A);
     elseif (isnumeric (A) && issquare (A))
       Ax = @(x) A * x;
     elseif (isa (A, "function_handle"))
       Ax = @(x) feval (A, x);
     else
       error ("cgs: A must be a square matrix or function");
     endif
 
